dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -570,98 +573,108 @@ public void run(){
-
+        final String endpointId = paramsMap.get("endpoint");
+        if (!UtilMethods.isSet(endpointId)) {
+            final Response.ResponseBuilder responseBuilder = Response.status(HttpStatus.SC_BAD_REQUEST);
+            responseBuilder.entity(
+                    responseMessage
+                            .append("Error: ")
+                            .append("endpoint")
+                            .append(" is a required Field."));
             return responseBuilder.build();
         }
 
         try {
-            JSONObject jsonResponse = new JSONObject();
+            final JSONObject jsonResponse = new JSONObject();
+            final HttpSession session = httpServletRequest.getSession();
 
-            HttpSession session = httpServletRequest.getSession();
             //Verify if we have something set on the session
             if ( session.getAttribute( "integrityCheck_" + endpointId ) == null ) {
                 //And prepare the response
-                jsonResponse.put( "success", false );
-                jsonResponse.put( "message", "No checking process found for End point server [" + endpointId + "]" );
-            } else if ( session.getAttribute( "integrityThread_" + endpointId ) == null ) {
+                jsonResponse.put("success", false);
+                jsonResponse.put("message", "No checking process found for End point server [" + endpointId + "]");
+            } else if (session.getAttribute("integrityThread_" + endpointId) == null) {
                 //And prepare the response
-                jsonResponse.put( "success", false );
-                jsonResponse.put( "message", "No checking process found for End point server [" + endpointId + "]" );
+                jsonResponse.put("success", false);
+                jsonResponse.put("message", "No checking process found for End point server [" + endpointId + "]");
             } else {
-
                 //Search for the status on session
-                ProcessStatus status = (ProcessStatus) session.getAttribute( "integrityCheck_" + endpointId );
+                final ProcessStatus status = (ProcessStatus) session.getAttribute("integrityCheck_" + endpointId);
 
                 //And prepare the response
-                jsonResponse.put( "endPoint", endpointId );
-                if ( status == ProcessStatus.PROCESSING ) {
-
+                jsonResponse.put("endPoint", endpointId);
+                if (status == ProcessStatus.PROCESSING) {
                     //Get the thread associated to this endpoint and the integrity request id
-                    Thread runningThread = (Thread) session.getAttribute( "integrityThread_" + endpointId );
-                    String integrityDataRequestId = (String) session.getAttribute( "integrityDataRequest_" + endpointId );
-
+                    final Thread runningThread = (Thread) session.getAttribute( "integrityThread_" + endpointId );
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)